deepmd/nvnmd/utils/config.py

```diff
@@ -189,7 +219,31 @@ def save(self, file_name=None):
             file_name = self.save_path
         else:
             self.save_path = file_name
+        self.update_config()
         FioDic().save(file_name, self.config)
+    
+    def get_s_range(self, davg, dstd):
+        rmin = nvnmd_cfg.dscp['rcut_smth']
+        rmax = nvnmd_cfg.dscp['rcut']
+        ntype = self.dscp['ntype']
+        dmin = self.dscp['dmin']
+        #
+        smin = 1e6
+        smax = -1e6
+        for tt in range(ntype):
+            smin_ = -davg[tt, 0] / dstd[tt, 0]
+            smax_ = (r2s(dmin, rmin, rmax) - davg[tt, 0]) / dstd[tt, 0]
+            smin = smin_ if (smin_ < smin) else smin
+            smax = smax_ if (smax_ > smax) else smax
```

Comment:
This might be written as array operation:
```py
smin_ = -davg[:, 0] / dstd[:, 0]
smax_ = (r2s(dmin, rmin, rmax) - davg[:, 0]) / dstd[:, 0]
smin = np.min(smin_)
smax = np.max(smax_)

Please recheck it.
```

Reply:
Thank you very much. We have made changes.